Stream-Three/Unit14-Custom_user_and_email_authentication/auth_demo/accounts/models.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import unicode_literals
+
+from django.contrib.auth.models import AbstractUser, UserManager
+from django.db import models
+from django.utils import timezone
+
+# Create your models here.
+class AccountUserManager(UserManager):
+    def _create_user(self, username, email, password,
+                     is_staff, is_superuser, **extra_fields):
+        """
+       Creates and saves a User with the given username, email and password.
+       """
+        now = timezone.now()
+        if not email:
+            raise ValueError('The given username must be set')
+
+        email = self.normalize_email(email)
+        user = self.model(username=email, email=email,
+                          is_staff=is_staff, is_active=True,
+                          is_superuser=is_superuser,
+                          date_joined=now, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+
+        return user
+
+
+class User(AbstractUser):
+    # now that we've abstracted this class we can add any
+    # number of custom attribute to our user class
+
+    # in later units we'll be adding things like payment details!
```

Comment:
I think I'd be very happy with a simpler version of this lesson, thanks for working on this Richard!

And concerning comments, I'm a big supporter of all the three things mentioned:
- docstrings for what the function does
- regular comments for how the function does it (whenever that's not quite clear from the code itself)
- TODO comments for how the function should be doing it, but isn't yet